src/Core/tests/UnitTests/Views/BorderTests.cs

```diff
@@ -29,6 +32,55 @@ public void TestBorderPropagateBindingContext()
 			Assert.True(border.BindingContext == bindingContext);
 		}
 
+		[Fact]
+		public void TestStrokeShapeBindingContext()
+		{
+			var context = new object();
+
+			var parent = new Border
+			{
+				BindingContext = context
+			};
+
+			var strokeShape = new Rectangle();
+
+			parent.StrokeShape = strokeShape;
+
+			Assert.Same(context, ((Rectangle)parent.StrokeShape).BindingContext);
+		}
+
+		[Fact]
+		public void BorderStrokeShapeSubscribed()
+		{
+			var strokeShape = new RoundRectangle { CornerRadius = new CornerRadius(12) };
+			var border = new Border { StrokeShape = strokeShape };
+
+			bool fired = false;
+			border.PropertyChanged += (sender, e) =>
+			{
+				if (e.PropertyName == nameof(Border.StrokeShape))
+					fired = true;
+			};
+
+			strokeShape.CornerRadius = new CornerRadius(24);
+			Assert.True(fired, "PropertyChanged did not fire!");
+		}
+
+		[Theory(Skip = "Reviewing why is failing")]
+		[InlineData(typeof(Rectangle))]
+		[InlineData(typeof(Ellipse))]
+		public async Task BorderStrokeShapeDoesNotLeak(Type type)
+		{
+			var strokeShape = (Shape)Activator.CreateInstance(type);
+			var reference = new WeakReference(new Border { StrokeShape = strokeShape });
+
+			await Task.Yield();
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+
+			Assert.False(reference.IsAlive, "Border should not be alive!");
+		}
```

Comment:
We should figure out why this is failing, I think it means we introduced a memory leak?

Reviewing the code I think something needs to unset this call:
```csharp
SetInheritedBindingContext(visualElement, BindingContext);
```

For example:

https://github.com/dotnet/maui/blob/917f4497a7df1416a57814e5945b8bb284211c3b/src/Controls/src/Core/VisualElement.cs#L310

Reply:
Found the leak, setting the Parent. Applying some changes.